### ISSUE TITLE
Replace the `Spring Data GemFire` dependency with `Spring Data Geode`

### ISSUE THREAD

**Issue by tzolov** [spring-projects/spring-integration#2352]
## Bug report

Currently Spring Integration [adds a dependency](https://github.com/spring-projects/spring-integration/blob/5f509ac02ed615f5fae5e8ec21c46418e6cdfd06/build.gradle#L376
) on `Spring Data Gemfire` which in turn will pulls non-open-source Gemfire dependencies. The Gemfire jars are commercially licensed, served from a [commercial (non-public) Maven repository](http://gemfire.docs.pivotal.io/gemfire/getting_started/installation/obtain_gemfire_maven.html)

Therefore SI should replace the [Spring Data GemFire](https://projects.spring.io/spring-data-gemfire/) dependency by its OSS equivalent [Spring Data Geode](https://docs.spring.io/autorepo/docs/spring-data-geode/2.0.x/reference/html/).

**Comment by artembilan** [comment]
@jxblum, 

WDYT, please?

Thanks

**Comment by jxblum** [comment]
Hi @artembilan -

Thank you for inquiring.  @tzolov and I had a conversation about this.  I am not particularly opposed and it is probably a good idea in the long-term.  2 primary reasons...

1. First, Apache Geode is OSS, [licensed](https://github.com/apache/geode/blob/rel/v1.4.0/LICENSE) under Apache License 2.0.  Additional, Apache Geode, like Spring Data Geode, or rather Spring Data for Apache Geode, is [available from Maven Central](http://search.maven.org/#search%7Cga%7C1%7Cgeode) making the transitive dependencies for _Spring_ application project (e.g. Boot/Integration/etc) easily resolvable without the need to add additional repository declarations in a Maven POM or Gradle build file for Pivotal GemFire (whether that is https://repo.spring.io/releases (shhhhhh ;-) or from PivNet).  Pivotal legal would like to have users "check" the license acceptable box.

2. As of _Spring Data_ **Kay**, I have gone to great effort to effectively make _Spring Data GemFire_ and _Spring Data Geode_ "**interchangeable**".  See [here](https://spring.io/blog/2017/10/26/diff-q-spring-data-gemfire-spring-data-geode) for more details.

In a nutshell, all a user need do is change their dependency declaration from `org.springframework.data:spring-data-geode` to `org.springframework.data:spring-data-gemfire` to switch from Apache Geode to Pivotal GemFire (assuming the user/customer wants "commercial" support now) following the 2.0.x release line (i.e. SD Kay) and s/he/they are done.  The inverse is also true!  If the user/customer wants to switch from Pivotal GemFire back to Apache Geode (back to OSS), this should work as well, without incident.  This is my SDG^2 guarantee.

As such, I think following the OS path makes the most sense.

Let me know if you have additional questions, comments or concerns.

Thanks!

**Comment by artembilan** [comment]
Great! Thank you both!

I don't think that the current point version `5.0.x` is a good place to do this, so, I'm deferring the task about moving to the `spring-data-geode` until the next `5.1` .

Cheers!
Artem

**Comment by artembilan** [comment]
Closed in favor of https://jira.spring.io/browse/INT-4398

**Comment by garyrussell** [comment]
Perhaps we should go ahead and create a spring-integration-geode project right away (just a copy) and deprecate the spring-integration-gemfire module.

I don't think (long term) we want spring-integration-gemfire to use geode; it could be misleading.

**Comment by garyrussell** [comment]
I guess we can't really deprecate because we might have commercial customers, but I don't think it's a huge maintenance issue.

**Comment by artembilan** [comment]
Well, from the Spring Integration perspective it doesn't matter: you just replace `gemfire` dependency to the `geode` and everything works well. 

Also keep in mind that the package in the module is indeed `gemfire`. Plus all the components use `Gemfire` prefix in their names 😄

**Comment by jxblum** [comment]
Right, exactly what I was going to say, thanks @artembilan (see comment directly above).

**Comment by garyrussell** [comment]
Right, so we should do what data did (parallel projects).

**Comment by artembilan** [comment]
So, we promote `spring-integration-geode` to the core project and move `spring-integration-gemfire` to the extensions since `5.1`. Just for the consistency to keep non-Open Source support out of Open Source.

WDYT?

**Comment by jxblum** [comment]
Also note, based on my referenced [blog post](https://spring.io/blog/2017/10/26/diff-q-spring-data-gemfire-spring-data-geode) above, as of SD Kay, the package namespace for Pivotal GemFire and Apache Geode, and by extension _Spring Data GemFire/Spring Data Geode_, align on the `org.apache.geode` package namespace (never mind class names; those are not going to change).

This has made life easier for merge between the 2 independent "parallel" projects, as you say, which is not only true of data, but I have followed the same approach in [_Spring Session Data GemFire/Geode_](https://github.com/spring-projects/spring-session-data-geode/blob/2.0.0.RELEASE/spring-session-data-geode/src/integration-test/java/org/springframework/session/data/gemfire/ClientServerHttpSessionAttributesDeltaIntegrationTests.java#L171-L172) and will do the same on the [_Spring Boot Data GemFire/Geode_](https://github.com/spring-projects/spring-boot-data-geode) project as well.

**Comment by jxblum** [comment]
FYI... I usually always start with Spring Data Geode and then merge my changes ("_cherry-pick_") to Spring Data GemFire, and for the most part, that just works, ;-).

Anyway, let me know if I can be of further assistance guys.

Cheers!
